lib/ansible/module_utils/urls.py

```diff
@@ -752,6 +764,15 @@ def build_ssl_validation_error(hostname, port, paths, exc=None):
     raise SSLValidationError(' '.join(msg) % (hostname, port, ", ".join(paths)))
 
 
+def atexit_remove_file(filename):
```

Comment:
I can rework the [cleanup function](https://github.com/ansible/ansible/pull/56532/files#diff-2bbdfb98f5cef18b20f2bec519a4bfe3R109) I have in #56532 so you can use that if that is helpful.

Reply:
I think in the end we would need separate functions.  One for use in "core" and one for use by modules.  For now keeping it here I think is fine. We can DRY it up at a time when that becomes a necessity.

Reply:
Ok. Sounds good.